examples/js_binary/BUILD.bazel

```diff
@@ -327,3 +327,52 @@ diff_test(
 aspect_test_a_bin.bin_a_test(
     name = "aspect_bin_a_test",
 )
+
+####################################################
+# Use case 10
+# Show launching js_binary() indirectly from a sh_binary()
+
+write_file(
+    name = "write10_js",
+    out = "case10.js",
+    content = ["""\
+        console.log("js_binary: ", __filename);
+        process.exit(0);
+"""],
+)
+js_binary(
+    name = "bin10-js_binary",
+    entry_point = "case10.js",
+)
+
+write_file(
+    name = "write10_launch_sh",
+    out = "launch_case10.sh",
+    content = ["""\
+        echo "sh launcher: $0"
+
+        # --- begin runfiles.bash initialization v2 ---
+        # Copy-pasted from the Bazel Bash runfiles library v2.
+        set -uo pipefail; f=bazel_tools/tools/bash/runfiles/runfiles.bash
+        source "${RUNFILES_DIR:-/dev/null}/$f" 2>/dev/null || \
+        source "$(grep -sm1 "^$f " "${RUNFILES_MANIFEST_FILE:-/dev/null}" | cut -f2- -d' ')" 2>/dev/null || \
+        source "$0.runfiles/$f" 2>/dev/null || \
+        source "$(grep -sm1 "^$f " "$0.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+        source "$(grep -sm1 "^$f " "$0.exe.runfiles_manifest" | cut -f2- -d' ')" 2>/dev/null || \
+        { echo>&2 "ERROR: cannot find $f"; exit 1; }; f=; set -e
+        # --- end runfiles.bash initialization v2 ---
+
+        $(rlocation aspect_rules_js/examples/js_binary/bin10-js_binary.sh) -h
+"""],
+)
+
+sh_binary(
```

Comment:
Note that this is a `sh_binary` and not a `sh_test` like we probably want. But the bug doesn't occur with `sh_test`. Any idea why the bug wouldn't occur when testing?

I think if we leave it as `sh_binary` then we need to add another rule (which I forget) to run it as a test?

Reply:
yea, it is sh_binary specific since sh_test sets the TEST_SRCDIR env var so RUNFILES is determined that way

Reply:
if you run the sh_binary through a genrule it is not reproducible either as the environment is setup differently in that case. only `bazel run` directly on the `sh_binary` will reproduce

Reply:
> yea, it is sh_binary specific since sh_test sets the TEST_SRCDIR env var so RUNFILES is determined that way

Not really related to this PR... but can RUNFILES be determined the same everywhere? Doing it different for tests seems odd 🤔

Reply:
Yeah, that code could get simplified in the future. Tricky to do right now as we don't have great coverage for different platforms and all of the corner cases.